### PR TITLE
Update Test Matrix

### PR DESCRIPTION
**Pull request by loicteixeira**
- Add Wagtail `1.11` and `1.12` to the test matrix
- Defines rules on what goes into the test matrix *

*Every time I want to update the test matrix, I'm left confused regarding which combination of Python/Django/Wagtail to test against. This is an attempt at codifying this, feel free to comment.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -3,3 +3,20 @@
...
 
 We align our browser support targets with that of Wagtail. Have a look at the [official documentation](http://docs.wagtail.io/en/latest/contributing/developing.html).
+
+## Python/Django/Wagtail support
+
+Supporting Python versions as defined in `setup.py` classifiers.
+
+Supporting Django versions as defined in the Django project [supported versions](https://www.djangoproject.com/download/#supported-versions).
+
+Supporting Wagtail versions from `1.7`.
```
Comment (loicteixeira) [inline docs/README.md]: I couldn't find a proper list of Wagtail's supported versions. The [release process](http://docs.wagtail.io/en/latest/contributing/release_process.html) document seems to imply that only the current version and the previous one will receive bug and security fixes but it also mentions LTS and I couldn't find a definite list of such versions. Therefore leaving the arbitrary `1.7` we started with.

File: tox.ini [hunk 2]
```diff
@@ -24,11 +24,12 @@ basepython =
 deps =
     dj18: Django>=1.8.1,<1.9
     dj110: Django>=1.10,<1.11
-    dj111: Django>=1.11,<1.12
+    dj111: Django>=1.11,<2.0
```
Comment (thibaudcolas) [inline tox.ini]: Why not keep 1.12? I think I've understood Django won't have a 1.12, but it still seems more correct in that case.

File: docs/README.md (hunk 1 above)
Comment (thibaudcolas) [reply docs/README.md]: There is new documentation at https://github.com/wagtail/wagtail/blob/a9b7265784761d7bd9428b9721bfea9486370c30/docs/releases/upgrading.rst, which I think will be published on the docs site in the upcoming 1.12 release. It also contains supported Django and Python versions per release.

Based on this, I think we should amend our support target to be the same as Wagtail's. As of _today_, that means:

- 1.8 (LTS)
- 1.10 (latest -1)
- 1.11 (latest)

The 1.12 release is just around the corner (first RC already done), once it's out that would mean:

- 1.8 (LTS)
- 1.11 (latest - 1)
- 1.12 (latest, and LTS)

Practically, we also have great incentives to support versions used on _our_ projects that use WagtailDraftail, so I think we should just drop 1.7 and 1.9 for now and still support 1.10.

File: docs/README.md [hunk 3]
```diff
@@ -3,3 +3,20 @@
...
+
+### Which version combinations to include in Travis test matrix?
+
+In order to keep for CI build time from growing out of control, not all Python/Django/Wagtail combinations will be tested.
+
+Test as follow:
+- All supported Django/Wagtail combinations with Python `2.7` (since it's the only version in the `2.x` series we support).
+- All supported Django/Wagtail combinations with the latest supported Python version of the `3.x` series.
+- The latest supported Django/Wagtail combinations for the remaining Python versions.
```
Comment (thibaudcolas) [inline docs/README.md]: 👍

File: .travis.yml [hunk 4]
```diff
@@ -23,32 +26,32 @@ matrix:
      python: 2.7
    - env: TOXENV=py27-dj110-wt110
      python: 2.7
+   - env: TOXENV=py27-dj110-wt111
+     python: 2.7
+   - env: TOXENV=py27-dj111-wt17
+     python: 2.7
+   - env: TOXENV=py27-dj111-wt18
```
Comment (thibaudcolas) [inline .travis.yml]: This version combination is not supported by Wagtail (https://github.com/wagtail/wagtail/blob/a9b7265784761d7bd9428b9721bfea9486370c30/docs/releases/upgrading.rst).

File: .travis.yml [hunk 5]
```diff
@@ -57,8 +60,18 @@ matrix:
      python: 3.6
    - env: TOXENV=py36-dj110-wt110
      python: 3.6
+   - env: TOXENV=py36-dj110-wt111
+     python: 3.6
+   - env: TOXENV=py36-dj111-wt17
+     python: 3.6
+   - env: TOXENV=py36-dj111-wt18
```
Comment (thibaudcolas) [inline .travis.yml]: This version combination is not supported by Wagtail (https://github.com/wagtail/wagtail/blob/a9b7265784761d7bd9428b9721bfea9486370c30/docs/releases/upgrading.rst).

**Review by thibaudcolas — COMMENTED** [review]
Thank you for taking the time to formalise our support/testing targets!

Overall looks good, I added some comments based on a (new) piece of documentation you probably missed. Please find my comments on how to take this into account below.

Here is what the test matrix would look like:

```txt
py27-dj18-wt18
py27-dj18-wt110
py27-dj18-wt111
py27-dj110-wt18
py27-dj110-wt110
py27-dj110-wt111
py27-dj111-wt110
py27-dj111-wt111

py34-dj111-wt111
py35-dj111-wt111

py36-dj18-wt18
py36-dj18-wt110
py36-dj18-wt111
py36-dj110-wt18
py36-dj110-wt110
py36-dj110-wt111
py36-dj111-wt110
py36-dj111-wt111
```

File: tox.ini (hunk 2 above)
Comment (loicteixeira) [reply tox.ini]: Simply trying to match Wagtail setup as close as possible.

File: docs/README.md (hunk 1 above)
Comment (loicteixeira) [reply docs/README.md]: `1.12` has just been release so I'll use that. However my understanding is that there are never 2 LTS versions at the same time? I'll remove `1.8` from the test matrix for now but we can add it again if I'm wrong.

**Comment by loicteixeira** [comment]
I realise the test site uses Wagtail `1.9` which as off this PR we don't support. I'll push the corresponding changes tomorrow.

**Review by thibaudcolas — DISMISSED** [review]
💯

**Comment by loicteixeira** [comment]
☝️ Oh interesting, force pushing a branch (because you've rebased master onto it for example) will dismiss the review 🤔

**Pull request merged**